BookReader/plugins/plugin.vendor-fullscreen.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Toggles browser's native fullscreen mode if available
+ */
+
+
+(function(BR) {
+  var event_namespace = ".bookreader_vendor-fullscreen";
+
+  jQuery.extend(BR.defaultOptions, {
+    enableVendorFullscreenPlugin: true
+  });
+
+  BR.prototype.setup = (function(super_) {
+    return function(options) {
+      super_.call(this, options);
+
+      this.isVendorFullscreenActive = false;
+    };
+  })(BR.prototype.setup);
+
+  BR.prototype.getInitialMode = (function(super_) {
+    return function(params) {
+      var nextMode = super_.call(this, params);
+      if (this.isVendorFullscreenActive) {
+        nextMode = this.constMode1up;
+      }
+      return nextMode;
+    };
+  })(BR.prototype.getInitialMode);
+
+  BR.prototype.init = (function(super_) {
+    return function() {
+      super_.call(this);
+
+      if (BR.util.fullscreenAllowed()) {
```

Comment:
You can return early since the entire rest of this method is inside this `if` statement.